.github/workflows/upstream-dev-ci.yaml

```
@@ -41,7 +41,7 @@ jobs:
           conda list
       - name: Run Tests
         run: |
-          python -m pytest --verbose -rf > output-${{ matrix.python-version }}-log
+          python -m pytest -rf | tee output-${{ matrix.python-version }}-log
```

Enable `pipefail`:

```suggestion
          set -euo pipefail
          python -m pytest -rf | tee output-${{ matrix.python-version }}-log
```

---

Thanks for catching this. Assuming the shell is `bash`, we don't need `-e` (exit the script if the pipeline failed) or `-u` (undefined variables are a error):
```suggestion
          set -o pipefail
          python -m pytest -rf | tee output-${{ matrix.python-version }}-log
```